### PR TITLE
fix(developer): getFontFamily now returns null if ttfMeta.promise errors

### PR DESCRIPTION
**Pull request by markcsinclair**
There are few conditions where ttfMeta.promise will return without a font family, instead it errors, so getFontFamily should catch this and return null

A test is provided for the corresponding keyboard info compiler message (ERROR_FontFileCannotBeRead).

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc-keyboard-info/src/font-family.ts [hunk 1]
```diff
@@ -16,7 +16,12 @@ export async function getFontFamily(source: Uint8Array) {
   }
 
   const buffer = Buffer.from(source);
-  const font = await ttfMeta.promise(buffer);
+  let font = null;
+  try {
+    font = await ttfMeta.promise(buffer);
+  } catch(e) {
+    return null;
```
Comment (mcdurdin) [inline developer/src/kmc-keyboard-info/src/font-family.ts:23]: It might be nice to capture the error string in the compiler error?

**Review by mcdurdin — APPROVED** [review]
LGTM

File: developer/src/kmc-keyboard-info/src/font-family.ts (hunk 1 above)
Comment (markcsinclair) [reply developer/src/kmc-keyboard-info/src/font-family.ts:23]: I'll open a new fix for this.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.288-beta](https://keyman.com/downloads/releases/beta/17.0.288)

**Pull request merged**